### DISCUSSION TITLE
Input form components

### DISCUSSION THREAD

**Discussion by woodseowl** [CornellCustomDev/CD-LaravelStarterKit#54] [Ideas]
We want a standard set of input components for forms that meet Custom Development requirements, including all of the management around them like error presentation. We need an example page like https://iws-preview.hosting.cornell.edu/ama39/cssf/forms.html so we can demonstrate what has been implemented.

This discussion is a place to capture ideas and contributions.

First question: what existing projects would be good to start from as examples?

**Comment by inaydich** [comment]
I would suggest to create a google spreadsheet with the list of possible components and the second column can have a list of the applications where example can be look for. It could be several apps listed and we might want to combine some code to create the best possible version.

**Reply by woodseowl** [reply]
Here is a rough start: https://docs.google.com/spreadsheets/d/15_w1yT7MOkJOjl_MKZhZq35JTDpjhTGBPEBvAAH3BVQ/edit?usp=sharing. Please improve on it!

Also, I don't have an Cornell workspace for this and I'm not sure if there's someplace else it's supposed to be.

**Reply by richmarisa** [reply]
We will need a few adjustments in version numbers: PHP 8.2, livewire 3, likely more.  Should I start a PR for this?

**Reply by woodseowl** [reply]
I think there is complexity around the Livewire 3 dependency for how the Starter Kit can be used. Could we wait until we see what comes of the comparisons conversation?

**Comment by richmarisa** [comment]
Livewire Volt requires livewire ^3,.0 Laravel ^10.17, Laravel/folio ^1.1, mockery/mockery ^1.6, orchestra/testbench ^8.6.2, pestphp/perst ^2.95, phpstan/phpstan ^1.10

From: Eric Woods ***@***.***>
Date: Monday, October 16, 2023 at 4:07 PM
To: CU-CommunityApps/CD-LaravelStarterKit ***@***.***>
Cc: Richard J. Marisa ***@***.***>, Comment ***@***.***>
Subject: Re: [CU-CommunityApps/CD-LaravelStarterKit] Input form components (Discussion #54)

I think there is complexity around the Livewire 3 dependency for how the Starter Kit can be used. Could we wait until we see what comes of the comparisons conversation?

**Reply by woodseowl** [reply]
Yes, gotcha on those dependency. I'm just saying can we wait for that part and push it out when we have the components that are going to use it?

I'm asking because I have some work on PhoneNumbers, LDAP, and CU Auth library code that could all be useful to try out as packages on some existing sites. But if we make the move now to Laravel 10 + Livewire 3, I won't be able to apply them because the sites aren't ready for one or the other of those upgrades.

**Reply by woodseowl** [reply]
Creating a PR now is low impact. It just may need to get updated by the time work on components is ready to merge. And depending on where things are at by then we may need to start talking about how we manage major releases / breaking changes.

**Comment by richmarisa** [comment]
Sure, that’s fine.

From: Eric Woods ***@***.***>
Date: Monday, October 16, 2023 at 4:35 PM
To: CU-CommunityApps/CD-LaravelStarterKit ***@***.***>
Cc: Richard J. Marisa ***@***.***>, Comment ***@***.***>
Subject: Re: [CU-CommunityApps/CD-LaravelStarterKit] Input form components (Discussion #54)

Yes, gotcha on those dependency. I'm just saying can we wait for that part and push it out when we have the components that are going to use it?

I'm asking because I have some work on PhoneNumbers, LDAP, and CU Auth library code that could all be useful to try out as packages on some existing sites. But if we make the move now to Laravel 10 + Livewire 3, I won't be able to apply them because the sites aren't ready for one or the other of those upgrades.

**Comment by woodseowl** [comment]
I think the work on PR #62 is largely taking this conversation forward, so I'm going to close this as now out of date.
